progression/src/main/java/me/mykindos/betterpvp/progression/tree/mining/event/ProgressionMiningEvent.java

```diff
@@ -0,0 +1,26 @@
+package me.mykindos.betterpvp.progression.tree.mining.event;
+
+import lombok.Getter;
+import lombok.Setter;
+import me.mykindos.betterpvp.core.framework.events.CustomEvent;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+
+import java.util.function.LongUnaryOperator;
+
+@Getter
+@Setter
+public class ProgressionMiningEvent extends CustomEvent {
+
+    Player player;
+
+    Block block;
+
+    LongUnaryOperator experienceModifier;
+
+    public ProgressionMiningEvent(Player player, Block block, LongUnaryOperator experienceModifier) {
```

Comment:
I don't think we need a `ProgressionMiningEvent` if it's just a relay for `BlockBreakEvent` with an experience modifier. If its sole purpose was for changing experience we can probably come up with a different way of adding experience modifiers by maybe attaching them to the block in the mining service. However, for this case, we already handle experience for broken fields ores in [MiningListener](https://github.com/Mykindos/BetterPvP/blob/c95087de5c4f9234f88853d13b415a2a26190179/clans/src/main/java/me/mykindos/betterpvp/clans/progression/listener/MiningListener.java#L28).

Reply:
kind of makes sense, though this implementation feels easier for me to understand

Reply:
Removed